bitbot/links.go

```diff
@@ -17,25 +18,31 @@ var URLReaderTrigger = NamedTrigger{
 	Action: func(irc *hbot.Bot, m *hbot.Message) bool {
 		resp := lookupPageTitle(m.Content)
 		if resp != "" {
-			irc.Reply(m, lookupPageTitle(m.Content))
+			irc.Reply(m, resp)
 		}
 		return true
 	},
 }
 
+var lastTimeTitleLookup = time.Date(
+        1970, 1, 1, 0, 0, 0, 0, time.UTC)
+
 func isURL(message string) bool {
 	return xurls.Strict().MatchString(message)
 }
 
 func lookupPageTitle(message string) string {
+	if time.Now().Sub(lastTimeTitleLookup).Minutes() < 2 {
```

Comment:
This function seems to put a space of two minutes between _any two URL lookups_, which is not what we were shooting for. We were trying to cache the URL title if it had been recently looked up and return that back to chat without performing unnecessary queries.